packages/story-editor/src/components/library/panes/pageTemplates/dropDownMenu.js

```diff
@@ -0,0 +1,224 @@
+          value: MENU_OPTIONS.EDIT,
+        },
+        {
+          label: __('Delete Template', 'web-stories'),
+          value: MENU_OPTIONS.DELETE,
+        },
+      ],
+    },
+  ];
+
+  const [showDeleteDialog, setShowDeleteDialog] = useState(false);
+  const [showRenameDialog, setShowRenameDialog] = useState(false);
+  const moreButtonRef = useRef();
+
+  const onClose = useCallback(() => {
+    onMenuCancelled();
+    moreButtonRef.current?.focus();
+    // Activeness of the parent is lost then due to blurring, we need to set it active "manually"
+    // to focus back on the original button.
+    setParentActive();
+  }, [onMenuCancelled, setParentActive]);
+
+  const handleCurrentValue = (evt, value) => {
+    evt.stopPropagation();
+    onMenuSelected();
+    switch (value) {
+      case MENU_OPTIONS.EDIT:
+        setShowRenameDialog(true);
+        break;
+      case MENU_OPTIONS.DELETE:
+        setShowDeleteDialog(true);
+        break;
+      default:
+        break;
+    }
+  };
+
+  // On Delete dialog closing.
+  const onDeleteDialogClose = useCallback(() => {
+    setShowDeleteDialog(false);
+    onClose();
+  }, [setShowDeleteDialog, onClose]);
+
+  // On Rename dialog closing.
+  const onRenameDialogClose = useCallback(() => {
+    setShowRenameDialog(false);
+    onClose();
+  }, [setShowRenameDialog, onClose]);
+
+  const listId = useMemo(() => `list-${uuidv4()}`, []);
+  const buttonId = useMemo(() => `button-${uuidv4()}`, []);
+
+  // Keep icon and menu displayed if menu is open (even if user's mouse leaves the area).
+  return (
+    <MenuContainer>
+      {moreButtonRef.current && (
+        <DropDownKeyEvents target={moreButtonRef.current} />
+      )}
+      <MoreButton
+        ref={moreButtonRef}
+        onClick={onMenuOpen}
+        aria-label={__('More', 'web-stories')}
+        aria-pressed={isMenuOpen}
+        aria-haspopup
+        aria-expanded={isMenuOpen}
+        aria-owns={isMenuOpen ? listId : null}
+        id={buttonId}
+        $display={display}
+        tabIndex={display || isMenuOpen ? 0 : -1}
+      >
+        <Icons.DotsFillSmall />
+      </MoreButton>
+      {(display || isMenuOpen) && (
+        <Popup
+          anchor={moreButtonRef}
+          placement={Placement.BottomStart}
+          isOpen={isMenuOpen}
+        >
+          <DropDownContainer>
+            <Menu
+              parentId={buttonId}
+              listId={listId}
+              handleMenuItemSelect={handleCurrentValue}
+              groups={groups}
+              onDismissMenu={onClose}
+              hasMenuRole
+              menuStylesOverride={menuStylesOverride}
+            />
+          </DropDownContainer>
+        </Popup>
+      )}
+      {showDeleteDialog && (
+        <DeleteDialog onClose={onDeleteDialogClose} onDelete={onDelete} />
+      )}
+      {showRenameDialog && (
+        <NameDialog
+          onClose={onRenameDialogClose}
+          onUpdateName={(name) => {
```

Comment:
```suggestion
          onSave={(name) => {
```

There's no `onUpdateName` prop on the component anymore.

Right now, because of this, renaming an existing page template does not work due to a JS error.

Reply:
The existing Karma and e2e tests did not catch this, so it would be great if you could update them so that they would catch an error like this.